### PR TITLE
Removes antimagic from holymelon

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request

Deletes antimagic component from holymelons

## Why It's Good For The Game

Holymelons were becoming overly centralizing to the antags that they countered with pocket antimagic, to the point botanists were growing it every single round just to metagame said antagonists.

Holymelons still contain holy water and thus are still a viable way to deal with cult without a chaplain.

## Changelog
:cl: Yakumo Chen
balance: removed antimagic component from holymelon
/:cl:

### CODE REVIEW

**Comment by Sishen1542** [comment]
why not just make them absorb a single magic hit instead of removing it entirely?

**Comment by YakumoChen** [comment]
because there's no point when you can make infinite melons
you can also make a pocket sized bag that holds 100 of them, so

**Comment by Sishen1542** [comment]
> because there's no point when you can make infinite melons

:thinking: 

hand only and with charges works fine on /tg/, so why remove the feature entirely on citadel? just port the /tg/ fixes

**Comment by YakumoChen** [comment]
You're typically only taking a hit from one spell at a time, so you only ever really need one charge. When you have a bag that can old literally 100 melons, the point becomes moot when you can just refresh your charge every time you get hit.

There isn't enough of a downside to outweigh the overwhelming centralization that becomes apparent when everyone can get antimagic from one item.

**Comment by Poojawa** [comment]
This isn't tg, we don't balance to tg's requirements 

I, for one, support this. because it's just lame that an antagonist's primary means of being different from just organized traitors is completely shut down by a single, easily obtainable resource. There's no counterplay.

**Comment by necromanceranne** [comment]
I suppose I should ask since I don't keep up to date on cit's balance changes, but did you guys happen to mirror this change over from /tg/? https://github.com/tgstation/tgstation/pull/41739

More specifically the holymelon bit rather than the antimagic change, since this pr actually caused a lot of messy code changes that broke antimagic for a bit. The melon change was pretty solid though, but it doesn't really fix the problem of holymelons being mass produced to counter cultists and wizards.

That said, I'd consider that cultists are, by /tg/ standards, fairly strong antags. I couldn't say what they're like on cit as I've never played cit with high enough population for cult. But tbh the antag always seemed to justify melon's existance. For wizard, it was kind of a big middle finger to their existence but given that not everyone takes the melons and there are some spells and gear that bypass antimagic (summon lesser guns, mutate, smoke, shapechange and mjolnir for example) meant wizards were still largely okay.

File: code/modules/hydroponics/grown/melon.dm [hunk 1]
```diff
@@ -58,6 +58,7 @@
 	wine_power = 70 //Water to wine, baby.
 	wine_flavor = "divinity"
 
-/obj/item/reagent_containers/food/snacks/grown/holymelon/Initialize()
-	. = ..()
-	AddComponent(/datum/component/anti_magic, TRUE, TRUE) //deliver us from evil o melon god
```
Comment (Trilbyspaceclone) [inline code/modules/hydroponics/grown/melon.dm:63]: ```AddComponent(/datum/component/anti_magic, FALSE , TRUE)```
So they keep their holy. Holy only protects against Watchers as well as Juggernauts 90 attack spell. As far as I know making them still weaker to stop cult hardstuns.

**Comment by Thalpy** [comment]
I know r4d6 was changing melons to be a % reduction rather than a binary, though they had a little trouble getting things to work. I could check up on them to see how they got along with it as a potential middleground?

**Comment by YakumoChen** [comment]
I do not believe any solution beyond full removal will deter the current abuse of holymelons to meta or counter cult/wizard.
 Pooj is right, shutting down or roadblocking the main abilities of an antag with no counterplay, with an easily spammed resource is bad design.

Alternatives I would consider solely would be to put an antimagic component in the metabolize process of holy water, while also returning the full brain damage effect of having holy water in your system, along with magic abilities also causing further negative effects like purging the water out of you or doing damage. It would be better counterplay compared to just having an item that completely negates antag stuff. It would still be abused though so even then I doubt id really consider it an alternative.

**Review by silicons — APPROVED** [review]
we'll readd this when antimagic/magic dampening is reworked to be able to be partial rather than absolute.

**Pull request merged**